impl/src/main/java/io/kroxylicious/testing/kafka/invm/InVMKafkaCluster.java

```diff
@@ -131,14 +157,14 @@ private KafkaConfig buildBrokerConfig(KafkaClusterConfig.ConfigHolder c, Path te
     }
 
     @Override
-    public void start() {
+    public synchronized void start() {
         // kraft mode: per-broker: 1 external port + 1 inter-broker port + 1 controller port + 1 anon port
         // zk mode: per-cluster: 1 zk port; per-broker: 1 external port + 1 inter-broker port + 1 anon port
         try (var preallocator = new ListeningSocketPreallocator()) {
-            externalPorts = preallocator.preAllocateListeningSockets(clusterConfig.getBrokersNum());
-            anonPorts = preallocator.preAllocateListeningSockets(clusterConfig.getBrokersNum());
-            interBrokerPorts = preallocator.preAllocateListeningSockets(clusterConfig.getBrokersNum());
-            controllerPorts = allocateControllerPorts(clusterConfig, preallocator);
+            Utils.putAllListEntriesIntoMapKeyedByIndex(preallocator.preAllocateListeningSockets(clusterConfig.getBrokersNum()), externalPorts);
+            Utils.putAllListEntriesIntoMapKeyedByIndex(preallocator.preAllocateListeningSockets(clusterConfig.getBrokersNum()), anonPorts);
+            Utils.putAllListEntriesIntoMapKeyedByIndex(preallocator.preAllocateListeningSockets(clusterConfig.getBrokersNum()), interBrokerPorts);
+            Utils.putAllListEntriesIntoMapKeyedByIndex(allocateControllerPorts(preallocator, clusterConfig), controllerPorts);
         }
 
         kafkaEndpoints = new KafkaClusterConfig.KafkaEndpoints() {
```

Comment:
I feel that `KafkaEndpoints` needs refactoring into something better.   I wondering between:

* turning it into a concrete class and give it the responsibility/ownership for the externalPorts/anonPorts fields
* or, having TestcontainersKafkaCluster|InVMKafkaCluster implement the KafkaEndpoints interface themselves.

I'm inclined to make that change in a separate PR.

Reply:
I opened a separate PR for this #109